code/modules/mining/drilling/drill.dm

```diff
@@ -85,7 +85,8 @@
 		harvesting.has_resources = FALSE
 		harvesting.resources = null
 		resource_field -= harvesting
-		harvesting = pick(resource_field)
+		if(resource_field.len)
+			harvesting = pick(resource_field)
```

Comment:
Why adding if(resource_field.len) check when while(resource_field.len && !harvesting.resources) already does it ?

Reply:
For this reason:

```
resource_field -= harvesting
harvesting = pick(resource_field)
```


### `resource_field -= harvesting`